pilot/pkg/config/kube/crdclient/client.go

```diff
@@ -124,20 +177,19 @@ func (cl *Client) Run(stop <-chan struct{}) {
 	t0 := time.Now()
 	scope.Info("Starting Pilot K8S CRD controller")
 
-	go func() {
-		if !cache.WaitForCacheSync(stop, cl.HasSynced) {
-			scope.Error("Failed to sync Pilot K8S CRD controller cache")
-			return
-		}
-		scope.Info("Pilot K8S CRD controller synced ", time.Since(t0))
-		cl.queue.Run(stop)
-	}()
-
+	if !cache.WaitForCacheSync(stop, cl.informerSynced) {
+		scope.Error("Failed to sync Pilot K8S CRD controller cache")
+		return
+	}
+	cl.SyncAll()
+	cl.initialSync.Store(true)
+	scope.Info("Pilot K8S CRD controller synced ", time.Since(t0))
+	cl.queue.Run(stop)
 	<-stop
```

Comment:
nit: probably don't need to wait on `stop` since the `Run` is on this goroutine.

Reply:
you are right, let me fix other places as well

Reply:
done